### PR TITLE
Added tests for edge cases

### PR DESCRIPTION
**Pull request by peter-lawrey**
This patch **adds 10 new JUnit tests (+3 CPU-layout fixtures)** to harden the Java Thread Affinity project against edge cases.  
It touches **_only_ the `src/test` tree** – no production code is modified.

| Test class | Purpose |
|------------|---------|
| **`AffinityLockDumpLocksTest`** | Verifies `dumpLocks()` lists all threads currently holding locks. |
| **`AffinityLockReleaseTest`** | Ensures `AffinityLock.release()` restores the mask to `BASE_AFFINITY`. |
| **`AffinityResetToBaseAffinityTest`** | Confirms `Affinity.resetToBaseAffinity()` resets the calling thread correctly. |
| **`AffinityThreadFactoryTest`** | Checks that threads created by `AffinityThreadFactory` each bind to a distinct CPU (Linux only). |
| **`CpuInfoLayoutMappingTest`** | Regression test for parsing `/proc/cpuinfo` into a correct logical-to-physical map. |
| **`UtilitiesTest`** | Edge-case conversions for `Utilities.toHexString` and `toBinaryString` with high/empty bits. |
| **`VanillaCpuLayoutPropertiesParseTest`** | Parses sample `.properties` layouts (i3, dual-Xeon, dual-E5405) and validates counts. |

*(Three helper classes extend `BaseAffinityTest` to reuse temporary-dir handling and Linux checks.)*

 Improves confidence in affinity-lock lifecycle, CPU-layout parsing, and utility helpers.

### CODE REVIEW

File: affinity/src/test/java/net/openhft/affinity/impl/VanillaCpuLayoutPropertiesParseTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package net.openhft.affinity.impl;
+
+import org.junit.Test;
+
+import java.io.InputStream;
+
+import static org.junit.Assert.assertEquals;
+
+public class VanillaCpuLayoutPropertiesParseTest {
```
Comment (tgd) [inline affinity/src/test/java/net/openhft/affinity/impl/VanillaCpuLayoutPropertiesParseTest.java:9]: We should cover Ryzen and Epyc here

**Review by tgd — APPROVED** [review]

File: affinity/src/test/java/net/openhft/affinity/impl/VanillaCpuLayoutPropertiesParseTest.java (hunk 1 above)
Comment (peter-lawrey) [reply affinity/src/test/java/net/openhft/affinity/impl/VanillaCpuLayoutPropertiesParseTest.java:9]: The CPU model here is illustrative, we could change the names of the test.

Comment (tgd) [reply affinity/src/test/java/net/openhft/affinity/impl/VanillaCpuLayoutPropertiesParseTest.java:9]: Got it, the properties files are cpuinfo dumps IIRC - it would prove that the parser handles the latest layouts (I assume it should just work)
